linera-service/src/database_tool.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use linera_service::storage::StorageConfig;
+use linera_views::common::CommonStoreConfig;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(name = "Clear database", about = "A tool for cleaning up a database")]
+struct DatabaseToolOptions {
+    /// Subcommands. Acceptable values are run and generate.
+    #[structopt(subcommand)]
+    command: DatabaseToolCommand,
+}
+
+#[derive(StructOpt)]
+enum DatabaseToolCommand {
+    /// Subcommands. Acceptable values are delete_all, delete_single, initialize
+
+    /// Delete all the entries of the database
+    #[structopt(name = "delete_all")]
+    DeleteAll {
+        /// Storage configuration for the blockchain history.
+        #[structopt(long = "storage")]
+        storage_config: String,
+
+        /// The maximal number of simultaneous queries to the database
+        #[structopt(long)]
+        max_concurrent_queries: Option<usize>,
+
+        /// The maximal number of simultaneous stream queries to the database
+        #[structopt(long, default_value = "10")]
+        max_stream_queries: usize,
+
+        /// The maximal number of entries in the storage cache.
+        #[structopt(long, default_value = "1000")]
+        cache_size: usize,
+    },
+
+    /// Delete a single table from the database
+    #[structopt(name = "delete_single")]
+    DeleteSingle {
+        /// Storage configuration for the blockchain history.
+        #[structopt(long = "storage")]
+        storage_config: String,
+
+        /// The maximal number of simultaneous queries to the database
+        #[structopt(long)]
+        max_concurrent_queries: Option<usize>,
+
+        /// The maximal number of simultaneous stream queries to the database
+        #[structopt(long, default_value = "10")]
+        max_stream_queries: usize,
+
+        /// The maximal number of entries in the storage cache.
+        #[structopt(long, default_value = "1000")]
+        cache_size: usize,
+    },
+
+    /// Delete a single table from the database
+    #[structopt(name = "existence")]
+    Existence {
```

Comment:
Commands are supposed to be verbs. Maybe `Check` or `CheckTable` ?

Reply:
Ok, renamed as `CheckExistence`.